flink-ml-lib/src/test/java/org/apache/flink/ml/feature/KBinsDiscretizerTest.java

```diff
@@ -0,0 +1,296 @@
+                    Row.of(Vectors.dense(7, 10, 0)),
+                    Row.of(Vectors.dense(10, 10, 0)),
+                    Row.of(Vectors.dense(13, 10, 3)));
+
+    private static final List<Row> TEST_INPUT =
+            Arrays.asList(
+                    Row.of(Vectors.dense(-1, 0, 0)),
+                    Row.of(Vectors.dense(1, 1, 1)),
+                    Row.of(Vectors.dense(1.5, 1, 2)),
+                    Row.of(Vectors.dense(5, 2, 3)),
+                    Row.of(Vectors.dense(7.25, 3, 4)),
+                    Row.of(Vectors.dense(13, 4, 5)),
+                    Row.of(Vectors.dense(15, 4, 6)));
+
+    private static final double[][] UNIFORM_MODEL_DATA =
+            new double[][] {
+                new double[] {1, 5, 9, 13},
+                new double[] {Double.MIN_VALUE, Double.MAX_VALUE},
+                new double[] {0, 1, 2, 3}
+            };
+
+    private static final List<Row> UNIFORM_OUTPUT =
+            Arrays.asList(
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(0, 0, 1)),
+                    Row.of(Vectors.dense(0, 0, 2)),
+                    Row.of(Vectors.dense(1, 0, 2)),
+                    Row.of(Vectors.dense(1, 0, 2)),
+                    Row.of(Vectors.dense(2, 0, 2)),
+                    Row.of(Vectors.dense(2, 0, 2)));
+
+    private static final List<Row> QUANTILE_OUTPUT =
+            Arrays.asList(
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(1, 0, 0)),
+                    Row.of(Vectors.dense(2, 0, 0)),
+                    Row.of(Vectors.dense(2, 0, 0)),
+                    Row.of(Vectors.dense(2, 0, 0)));
+
+    private static final List<Row> KMEANS_OUTPUT =
+            Arrays.asList(
+                    Row.of(Vectors.dense(0, 0, 0)),
+                    Row.of(Vectors.dense(0, 0, 1)),
+                    Row.of(Vectors.dense(0, 0, 2)),
+                    Row.of(Vectors.dense(1, 0, 2)),
+                    Row.of(Vectors.dense(1, 0, 2)),
+                    Row.of(Vectors.dense(2, 0, 2)),
+                    Row.of(Vectors.dense(2, 0, 2)));
+
+    private static final double TOLERANCE = 1e-7;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+        trainTable = tEnv.fromDataStream(env.fromCollection(TRAIN_INPUT)).as("input");
+        testTable = tEnv.fromDataStream(env.fromCollection(TEST_INPUT)).as("input");
+    }
+
+    @SuppressWarnings("all")
```

Comment:
nit: it might be better to replace this with the detailed warnings you want to suppress.

Reply:
Thanks for pointing this out. I have updated it with `@SuppressWarnings("unchecked, ConstantConditions")`